packages/web/modals/profile.tsx

```diff
@@ -0,0 +1,442 @@
+          <div className="flex items-center gap-1.5">
+            <Image
+              src="/icons/profile-wallet.svg"
+              alt="Osmo icon"
+              width={24}
+              height={24}
+            />
+            <p className="subtitle1 tracking-wide text-osmoverse-300">
+              {t("profile.wallet")}
+            </p>
+          </div>
+
+          <div className="flex justify-between 1.5xl:gap-4 1.5xs:flex-col">
+            <div className="flex gap-3">
+              <div className="h-12 w-12 shrink-0">
+                <Image
+                  alt="wallet-icon"
+                  src={navBarStore.walletInfo.logoUrl}
+                  height={48}
+                  width={48}
+                />
+              </div>
+
+              <div className="subtitle-1 tracking-wide">
+                <p>Cosmos</p>
+                <div className="flex items-center gap-2">
+                  <p title={address} className="text-osmoverse-100">
+                    {Bech32Address.shortenAddress(address, 12)}
+                  </p>
+                  <button
+                    title="Copy"
+                    onClick={onCopyAddress}
+                    className="group"
+                  >
+                    {hasCopied ? (
+                      <CheckMarkIcon
+                        classes={{
+                          container: "text-osmoverse-200",
+                        }}
+                      />
+                    ) : (
+                      <CopyIcon
+                        classes={{
+                          container: "w-[20px] h-[20px] text-osmoverse-200",
+                        }}
+                        isAnimated
+                      />
+                    )}
+                  </button>
+                </div>
+              </div>
+            </div>
+
+            <div className="flex items-center gap-3">
+              <ActionButton
+                title="Mintscan"
+                isLink
+                href={`https://www.mintscan.io/osmosis/account/${address}`}
+                target="blank"
+                className="group"
+                rel="noopener noreferrer"
+              >
+                <ExternalLinkIcon isAnimated />
+              </ActionButton>
+
+              <Drawer isOpen={isQROpen} onOpen={onOpenQR} onClose={onCloseQR}>
+                <DrawerButton>
+                  <ActionButton title="QR Code" className="group">
+                    <QRIcon isAnimated />
+                  </ActionButton>
+                </DrawerButton>
+
+                <DrawerContent>
+                  <DrawerOverlay />
+                  <DrawerPanel className="flex h-fit items-center justify-center pt-7 pb-7">
+                    <h6 className="mb-8">Cosmos</h6>
+                    <div className="mb-7 flex items-center justify-center rounded-xl bg-white-high p-3.5">
+                      <QRCode value={address} size={260} />
```

Comment:
We should try to use next/dynamic wherever possible, to keep the main bundle small. QRCode is a perfect example use case of using next/dynamic to explicitly keep it out of the main bundle.

https://github.com/osmosis-labs/osmosis-frontend/blob/c19969322b18a603992af37fd14d0cc749be5dd8/packages/web/modals/wallet-connect.tsx#L75

In this case, simply add 
`const QRCode = dynamic(() => import("qrcode.react"));`
At the top of the file, instead of the import statement.

Reply:
Also- it's totally saved us for stuff like lottie animations. We can still take advantage of those big powerful packages, without reducing time-to-interactive. Of course, the tradeoff in this case is the user sees some delay when opening the view. In this case, it's saved us 100+kb.

https://github.com/osmosis-labs/osmosis-frontend/blob/c19969322b18a603992af37fd14d0cc749be5dd8/packages/web/components/animation/bridge/index.tsx#L11